### ISSUE TITLE
exec plugin: invalid apiVersion "client.authentication.k8s.io/v1beta1"

### ISSUE THREAD

**Issue by deadNightTiger** [splunk/qbec#83]
My `$HOME/.kube/config`:
```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: "<snip>"
    server: "<snip>"
  name: kubernetes
contexts:
- context:
    cluster: kubernetes
    namespace: default
    user: oidc
  name: oidc@kubernetes
current-context: oidc@kubernetes
kind: Config
preferences: {}
users:
- name: oidc
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - get-token
      - --oidc-issuer-url=<snip>
      - --oidc-client-id=kubernetes
      - --oidc-extra-scope=email
      - --oidc-extra-scope=groups
      command: kubelogin
```

qbec's output (for any subcommand communicating with the cluster):
```
setting cluster to kubernetes
setting context to oidc@kubernetes
✘ exec plugin: invalid apiVersion "client.authentication.k8s.io/v1beta1"
```

Maybe it's time to update client-go?

**Comment by gotwarlost** [comment]
Thanks for reporting this issue. What is the server version of your server?

**Comment by gotwarlost** [comment]
Oh I misread this. This is happening when client go tries to invoke a local exec plugin, right?

**Comment by deadNightTiger** [comment]
Yes, I think so. The reason, I believe, is that qbec uses client-go v7, which corresponds to kubernetes 1.10, but this apiVersion only got promoted to beta in 1.11 (https://github.com/kubernetes/kubernetes/pull/64482).

**Comment by gotwarlost** [comment]
@deadNightTiger could you do me a favor? Please checkout the `up-client-go` branch, build qbec from source (`make`) and see if your issue is resolved.

Thanks in advance.

**Comment by deadNightTiger** [comment]
Yep, it works from the branch! Thanks for a quick reaction :+1:

**Comment by gotwarlost** [comment]
Thanks for the quick test. We'll make a release this weekend.

**Comment by gotwarlost** [comment]
fixed via https://github.com/splunk/qbec/commit/c963950aaf8cba8a6950067f84fbc32732990545 and available in `v0.9.0` release https://github.com/splunk/qbec/releases/tag/v0.9.0
